src/main/java/org/opentripplanner/street/search/state/StateEditor.java

```diff
@@ -196,6 +197,19 @@ public void incrementWalkDistance(double length) {
     child.walkDistance += length;
   }
 
+  public void incrementDrivenBatteryMeters(double length) {
+    if (length < 0) {
+      LOG.warn("A state's battery distance is being incremented by a negative amount.");
```

Comment:
I will have to check with the other developers but I would be totally unforgiving and throw an exception in this case. This indicates a bug elsewhere.

Reply:
The `incrementWalkDistance`,  `incrementTimeInSeconds`, `incrementWeight`  methods in the StateEditor class do it the same way(thwowing no exception. So probably, if we change this, we should change those methods too.

We could add exceptions like NegativeBatteryMeterIncrement, NegativeWalkDistanceIncrement...

But we don' t know really good the exception handling in OTP. 
So at least, I would prefer to create a new PR for that, if we should change that.